core/src/main/java/org/apache/gravitino/audit/AuditLogWriter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.audit;
+
+import java.util.Map;
+import org.apache.gravitino.listener.api.event.Event;
+
+/** Interface for writing the audit log. */
+public interface AuditLogWriter {
```

Comment:
How about define a default method for write to use formatter to format the event like 
```
Formatter<R> getFormatter();

void doWrite(R auditLog);

default void write(Event event) {
    doWrite(getFormatter().format(event))
}
```

Reply:
I'm fine with your idea, I think it is not a block issue. What I feel in the code is that it loses the type information for formatter in the code.